File: data/multilang_comparison/program_julia/src/func31.jl
include("program_julia_head.jl")
function func31(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   array28 = nothing
   if pCounter > 1
       array28 = vars.data[pCounter]
       pCounter -= 1
       array28.refC += 1
   else
       array28 = JArray(zeros(Int, 97), 97, 1, 28)
   end
   params0 = JArray_param{Int}()
   params0.size = 1
   push!(params0.data, array28)
   params0.size += 1
   array29 = func35(params0, loopsFactor);
   if (PATH0 & 1) != 0
      array32 = nothing
      if pCounter > 1
          array32 = vars.data[pCounter]
          pCounter -= 1
          array32.refC += 1
      else
          array32 = JArray(zeros(Int, 444), 444, 1, 32)
      end
      loopLimit11 = (50)/4 + 1
      for i in 1:loopLimit11 
         for i in 0:(array32.size-1) 
            array32.data[i+1] -= 1
         end
         for i in 1:array29.size
             if array29.data[i] == 29
                     return array29
            end
         end
      end
      array32.refC -= 1
      if array32.refC == 0
          empty!(array32.data)
          array32 = nothing
      end
   
   else 
      params1 = JArray_param{Int}()
      params1.size = 2
      push!(params1.data, array28)
      params1.size += 1
      push!(params1.data, array29)
      params1.size += 1
      array33 = func40(params1, loopsFactor);
      array33.refC -= 1
      if array33.refC == 0
          empty!(array33.data)
          array33 = nothing
      end
   end
   array28.refC -= 1
   if array28.refC == 0
       empty!(array28.data)
       array28 = nothing
   end
   return array29;
end

File: data/multilang_comparison/program_c/src/func31.c
#include "program_c.h" 
array_t* func31(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array28;
   if (pCounter > 0) {
      array28 = vars->data[--pCounter];
      array28->refC++;
      DEBUG_COPY(array28->id);
   } else {
      array28 = (array_t*)malloc(sizeof(array_t));
      array28->size = 97;
      array28->refC = 1;
      array28->id = 28;
      array28->data = (unsigned int*)malloc(array28->size*sizeof(unsigned int));
      memset(array28->data, 0, array28->size*sizeof(unsigned int));
      DEBUG_NEW(array28->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array28;
   array_t* array29 = func35(&params0, loopsFactor);
   DEBUG_RETURN(array29->id);
   free(params0.data);
   if(PATH0 & 1) {
      array_t* array32;
      if (pCounter > 0) {
         array32 = vars->data[--pCounter];
         array32->refC++;
         DEBUG_COPY(array32->id);
      } else {
         array32 = (array_t*)malloc(sizeof(array_t));
         array32->size = 444;
         array32->refC = 1;
         array32->id = 32;
         array32->data = (unsigned int*)malloc(array32->size*sizeof(unsigned int));
         memset(array32->data, 0, array32->size*sizeof(unsigned int));
         DEBUG_NEW(array32->id);
      }
      unsigned int loop11 = 0;
      unsigned int loopLimit11 = (50)/4 + 1;
      for(; loop11 < loopLimit11; loop11++) {
         for (int i = 0; i < array32->size; i++) {
            array32->data[i]--;
         }
         for (int i = 0; i < array29->size; i++) {
            if (array29->data[i] == 29) { 
               return array29;
            }
         }
      }
      array32->refC--;
      if(array32->refC == 0) {
         free(array32->data);
         free(array32);
         DEBUG_FREE(array32->id);
      }
   }
   else {
      array_t_param params1;
      params1.size = 2;
      params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
      params1.data[0] = array28;
      params1.data[1] = array29;
      array_t* array33 = func40(&params1, loopsFactor);
      DEBUG_RETURN(array33->id);
      free(params1.data);
      array33->refC--;
      if(array33->refC == 0) {
         free(array33->data);
         free(array33);
         DEBUG_FREE(array33->id);
      }
   }
   array28->refC--;
   if(array28->refC == 0) {
      free(array28->data);
      free(array28);
      DEBUG_FREE(array28->id);
   }
   return array29;
}

